numpy/ma/tests/test_core.py

```diff
@@ -4376,16 +4376,27 @@ def test_getitem(self):
                                    [1, 0, 0, 0, 0, 0, 0, 0, 1, 0])),
                           dtype=[('a', bool), ('b', bool)])
         # No mask
-        self.assertTrue(isinstance(a[1], MaskedArray))
+        assert_equal(type(a[1]), mvoid)
+        assert_equal(type(a[1,...]), MaskedArray)
+
         # One element masked
-        self.assertTrue(isinstance(a[0], MaskedArray))
+        assert_equal(type(a[0]), mvoid)
         assert_equal_records(a[0]._data, a._data[0])
         assert_equal_records(a[0]._mask, a._mask[0])
+
+        assert_equal(type(a[0,...]), MaskedArray)
+        assert_equal_records(a[0,...]._data, a._data[0,...])
+        assert_equal_records(a[0,...]._mask, a._mask[0,...])
+
         # All element masked
-        self.assertTrue(isinstance(a[-2], MaskedArray))
+        assert_equal(type(a[-2]), mvoid)
```

Comment:
The behaviour here hasn't changed, the test was just less strict than it should be